core/shared/src/main/scala/zio/stm/TRef.scala

```diff
@@ -142,18 +152,15 @@ final class TRef[A] private (
     updateAndGet(f orElse { case a => a })
 
   private[stm] def getOrMakeEntry(journal: Journal): Entry =
-    if (journal.containsKey(self)) journal.get(self)
-    else {
-      val entry = Entry(self, false)
-      journal.put(self, entry)
-      entry
-    }
+    journal.getOrElseUpdate(self, newEntry())
 
   private[stm] def unsafeGet(journal: Journal): A =
     getOrMakeEntry(journal).unsafeGet
 
   private[stm] def unsafeSet(journal: Journal, a: A): Unit =
     getOrMakeEntry(journal).unsafeSet(a)
+
+  private[this] val newEntry = () => Entry(self.asInstanceOf[TRef[A & AnyRef]])
```

Comment:
Is there a reason to define methods as `val func = () =>` vs `def func()`

Reply:
I found this out while looking at decompiled code.

The `getOrElseUpdate(key, newValue)` method on Scala's `HashMap`, has the second param as `=> A`. Since `=> A` is just syntactic sugar for `() => A`, this way we're not allocating or creating any closures when we call `getOrElseUpdate`.

Reply:
I had a hunch that this is the case, but was not sure. Good to know 👍